Sources/Sound.swift

```diff
@@ -118,19 +118,56 @@ open class Sound {
     /// Create a sound object.
     ///
     /// - Parameter url: Sound file URL.
-    public init?(url: URL) {
+    public convenience init?(url: URL) {
+        let constructor: ((Any) -> Player?) = { (URL) -> Player? in
+            do {
+                let player = try Sound.playerClass.init(contentsOf: url)
+                return player
+            } catch let error {
+                print("SwiftySound initialization error: \(error)")
+                return nil
+            }
+        }
+        self.init(object: url, constructor: constructor)
+    }
+    
+    /// Create a sound object.
+    ///
+    /// - Parameter data: Sound file Data.
+    public convenience init?(data: Data) {
+        let constructor: ((Any) -> Player?) = { (Data) -> Player? in
+            do {
+                let player = try Sound.playerClass.init(data: data)
+                return player
+            } catch let error {
+                print("SwiftySound initialization error: \(error)")
+                return nil
+            }
+        }
+        self.init(object: data, constructor: constructor)
+    }
+    
+    /// Create a sound object.
+    ///
+    /// - Parameter name: Sound name in asset catalog.
+    @available(iOS 9.0, *)
+    public convenience init?(name: String) {
+        guard let asset = NSDataAsset(name: name) else {
+            return nil
+        }
+        self.init(data: asset.data)
+    }
+    
+    private init?(object: Any, constructor: ((Any) -> Player?)) {
```

Comment:
I couldn't find a way to define a something like `constructor: ((Any) -> Player?) throws` to make this even simpler.